modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -419,7 +419,9 @@
   [driver [_ arg p]]
   (let [[offset quantiles] (percentile->quantile p)]
     (hsql/raw (format "APPROX_QUANTILES(%s, %s)[OFFSET(%s)]"
-                      (hformat/to-sql (sql.qp/->honeysql driver arg))
+                      (-> (sql.qp/->honeysql driver arg)
+                          (hformat/format :quoting (sql.qp/quote-style driver))
+                          first)
```

Comment:
The right way to do this would be to have this function return a reified `ToSql`, here's an example: https://github.com/metabase/metabase/blob/d2433696380fb43bb2ffaf5e1b2aabebd1354bc2/src/metabase/driver/postgres.clj#L186-L197

We don't need to use `hsql/raw` here. In the impl of `to-sql` then we can recursively apply `to-sql` to its args

Reply:
Made the change.